docs/async/async-overview.rst

```diff
@@ -0,0 +1,30 @@
+Async Overview
+==============
+
+Modern apps and services are expected to be responsive.
+
+* Client applications are expected to be always-on, always-connected, but keep the UI free for the user to interact with.
+* Services are expected to be able gracefully handle a heavy load.
+
+.NET provides the capability for apps and services to be responsive with easy-to-use, language-level asynchronous programming models in C#/VB and F#.  Although similar, the nuances of async programming in F# differs from that in C# and VB.
+
+Why Async?
```

Comment:
Why did we implement it or why it is needed at all? I think it would read better if the title is something along the lines of "Why intrinsic support for async?" or something similar.